### PR TITLE
Add support for TAU constant.

### LINKED ISSUE #12094: Support TAU constant as well as PI
Nowadays there is a [strong case for TAU](https://tauday.com/tau-manifesto).
Mathematicians and teachers are organically adopting it. _(any [khan academics](https://www.khanacademy.org/) here?)_

I just coded (discovering for the first time with immense delight this beautiful software that is godot) a simple script to rotate like a trackball a `Spatial` node from mouse inputs, and the code is easier to understand with whole rotations (TAU) as baseline instead of half-rotations (PI).

Here's the excerpt :

``` godot
var TAU = 2*PI

var t = node.get_transform()
var up = t.xform_inv(cameraUp)
t = t.rotated(up, tangent.x * TAU)
```

---

I'd suggest going as far as to add a deprecation notice to `PI`, suggesting `TAU`, but maybe it's too soon.

---

This is a very opinionated feature request, with a hint of a pedagogic stance, so I'm not sure you all will accept this. It's a breaking change, too, for people who defined TAU already.

But it is future-oriented and rationally justified ; waiting for the world to change is alwayss easy, rarely usseful.

_(I just half-quoted Professor Quirrel, so I'm not going to quote Gandhi afterwards, but you get the idea)_

**Let's see them thumbs ! What do you think ?**

### PR DESCRIPTION
**Pull request by Goutte**
Fixes #12094.
<sub>*bugsquad edit:* s/See/Fixes/</sub>

### CODE REVIEW

**Comment by Goutte** [comment]
Travis fails, but it looks unrelated.

File: modules/visual_script/visual_script_nodes.cpp [hunk 1]
```diff
@@ -1888,6 +1890,7 @@ void VisualScriptMathConstant::_bind_methods() {
 	BIND_ENUM_CONSTANT(MATH_CONSTANT_PI);
 	BIND_ENUM_CONSTANT(MATH_CONSTANT_2PI);
```
Comment (akien-mga) [inline modules/visual_script/visual_script_nodes.cpp]: Maybe we don't need 2PI if we have TAU?

Comment (Noshyaar) [reply modules/visual_script/visual_script_nodes.cpp]: I think we should relabel the "PIx2" to "TAU (PIx2)"?

Comment (Goutte) [reply modules/visual_script/visual_script_nodes.cpp]: I kept it for backwards compatibility. Maybe that would be a place for a deprecation warning ?

**Comment by HummusSamurai** [comment]
OP can be reworded as "Fixes #12094" to automate stuff.

File: modules/visual_script/visual_script_nodes.cpp (hunk 1 above)
Comment (bojidar-bg) [reply modules/visual_script/visual_script_nodes.cpp]: Break compatibility, we warned people that alpha 2 is not final anyway :wink:

Comment (akien-mga) [reply modules/visual_script/visual_script_nodes.cpp]: Bump.

Comment (Goutte) [reply modules/visual_script/visual_script_nodes.cpp]: I did the changes, but I'm having trouble rewriting my branch on github. I amended the commit locally, and rebased on the master. There is still the original commit on the remote branch `Goutte:feat-support-tau` on github and I'm not sure if I can rewrite its history.

Of course I could make a new branch and a new PR, but where's the fun in that ?

**Pull request merged**